Nos/Views/ProfileEdit/CreateUsernameWizard/ExternalNIP05/AlreadyHaveANIP05View.swift

```diff
@@ -0,0 +1,199 @@
+                                Task {
+                                    await verify(newValue)
+                                }
+                            }
+                            .onSubmit {
+                                Task {
+                                    await verify(usernameObserver.text)
+                                }
+                            }
+                    }
+                    if linkFailed {
+                        unableToLinkUsernameText()
+                    } else {
+                        unableToLinkUsernameText()
+                            .hidden()
+                    }
+
+                    Spacer(minLength: 0)
+                }
+
+                NavigationLink {
+                    ExcellentChoiceSheet(username: usernameObserver.text, isPresented: $isPresented)
+                } label: {
+                    if isVerifying {
+                        ZStack {
+                            ProgressView()
+                                .frame(height: .zero)
+                                .tint(Color.white)
+                            PlainText(.localizable.next)
+                                .hidden()
+                        }
+                    } else {
+                        PlainText(.localizable.next)
+                    }
+                }
+                .buttonStyle(BigActionButtonStyle())
+                .disabled(verified != true || isVerifying || invalidInput)
+                Spacer(minLength: 40)
+            }
+        }
+    }
+
+    private func unableToLinkUsernameText() -> some View {
+        WizardSheetDescriptionText(markdown: .localizable.nip05LinkFailed, tint: .red)
+            .font(.clarity(.medium, textStyle: .subheadline))
+            .lineSpacing(3)
+    }
+
+    private var invalidInput: Bool {
+        let input = usernameObserver.text
+        let emailRegEx = "[0-9a-z._-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+        let emailPred = NSPredicate(format: "SELF MATCHES %@", emailRegEx)
+        return emailPred.evaluate(with: input)
+    }
+
+    private var linkFailed: Bool {
+        verified == false
+    }
+
+    private func verify(_ username: String) async {
+        verified = nil
+
+        guard !username.isEmpty, let keyPair = currentUser.keyPair else {
+            return
+        }
+
+        let components = username.components(separatedBy: "@")
+
+        guard components.count == 2 else {
+            return
+        }
+
+        let localPart = components[0]
+        let domain = components[1]
+
+        guard let host = URL(string: "https://\(domain)/.well-known/nostr.json") else {
+            return
+        }
+
+        isVerifying = true
+
+        defer {
+            isVerifying = false
+        }
+
+        do {
+            verified = try await namesAPI.verify(username: localPart, host: host, keyPair: keyPair)
+        } catch {
+            Log.error(error.localizedDescription)
+            verified = false
```

Comment:
This line is redundant because of the above `defer`, right?

Reply:
the `defer` modifies `isVerifying` not `verified`